src/plugins/data_source_management/public/components/direct_query_data_sources_components/acceleration_creation/selectors/source_selector.tsx

```diff
@@ -0,0 +1,309 @@
+        {accelerationFormData.dataSource}
+      </EuiDescriptionListDescription>
+    </EuiDescriptionList>
+  );
+
+  const loadDataSource = () => {
+    setLoadingComboBoxes({ ...loadingComboBoxes, dataSource: true });
+    http
+      .get(`${DATACONNECTIONS_BASE}/dataSourceMDSId=${dataSourceMDSId ?? ''}`)
+      .then((res) => {
+        const isValidDataSource = res.some(
+          (connection: any) =>
+            connection.connector.toUpperCase() === 'S3GLUE' &&
+            connection.name === selectedDatasource
+        );
+
+        if (!isValidDataSource) {
+          notifications.toasts.addDanger(
+            `Received an invalid datasource in create acceleration flyout`
+          );
+        }
+      })
+      .catch((err) => {
+        // eslint-disable-next-line no-console
+        console.error(err);
+        notifications.toasts.addDanger(`failed to load datasources`);
+      });
+    setLoadingComboBoxes({ ...loadingComboBoxes, dataSource: false });
+  };
+
+  const loadDatabases = () => {
+    const dsCache = CatalogCacheManager.getOrCreateDataSource(
+      accelerationFormData.dataSource,
+      dataSourceMDSId
+    );
+
+    if (dsCache.status === CachedDataSourceStatus.Updated && dsCache.databases.length > 0) {
+      const databaseLabels = dsCache.databases.map((db) => ({ label: db.name }));
+      setDatabases(databaseLabels);
+    } else if (
+      (dsCache.status === CachedDataSourceStatus.Updated && dsCache.databases.length === 0) ||
+      dsCache.status === CachedDataSourceStatus.Empty
+    ) {
+      setDatabases([]);
+    }
+
+    setSelectedDatabase([]);
+    setTables([]);
+    setSelectedTable([]);
+    setAccelerationFormData({ ...accelerationFormData, database: '', dataTable: '' });
+  };
+
+  const loadTables = () => {
+    if (selectedDatabase.length > 0) {
+      let dbCache = {} as CachedDatabase;
+      try {
+        dbCache = CatalogCacheManager.getDatabase(
+          accelerationFormData.dataSource,
+          accelerationFormData.database,
+          dataSourceMDSId
+        );
+        if (dbCache.status === CachedDataSourceStatus.Updated && dbCache.tables.length > 0) {
+          const tableLabels = dbCache.tables.map((tb) => ({ label: tb.name }));
+          setTables(tableLabels);
+        } else if (
+          (dbCache.status === CachedDataSourceStatus.Updated && dbCache.tables.length === 0) ||
+          dbCache.status === CachedDataSourceStatus.Empty
+        ) {
+          setTables([]);
+        }
+      } catch (error) {
+        setTables([]);
+        notifications.toasts.addWarning('Your cache is outdated, refresh databases and tables');
+        // eslint-disable-next-line no-console
+        console.error(error);
+      }
+      setSelectedTable([]);
+      setAccelerationFormData({ ...accelerationFormData, dataTable: '' });
+    }
+  };
+
+  useEffect(() => {
+    loadDataSource();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
maybe move to top for file lever disabling?

Reply:
sure, just applied the changes

Reply:
fixed